### PR TITLE
[doc] Improve compatibility matrix notes

### LINKED ISSUE #1507: Kubernetes v1.18 support statement misleading
**What happened**:
On a Kubernetes v1.18 cluster installing _kube-state-metrics_ via the _kube-prometheus-stack_ Helm chart we followed the documentation and modified the chart image to use `v1.9.8` which errored due to a change to argument name (I'm not sure where this change is documented?). We then tried `v2.0.0-alpha.1` which appeared to work but it has a known issue with missing container request/limit metrics (https://github.com/kubernetes/kube-state-metrics/issues/1292) which is a major defect.

**What you expected to happen**:
The documentation to suggest using `v1.9.8` but point out that it's incompatible with the Helm chart. Don't recommend an alpha release with a major known issue.

**How to reproduce it (as minimally and precisely as possible)**:
Run `v2.0.0-alpha.1`.

**Anything else we need to know?**:
n/a

**Environment**:
- kube-state-metrics version: `v2.0.0-alpha.1`
- Kubernetes version (use `kubectl version`): v1.18.9
- Cloud provider or hardware configuration: AWS EKS
- Other info: n/a

### PR DESCRIPTION
**Pull request by stevehipwell**
**What this PR does / why we need it**:
This PR tidies up the notes on the compatibility matrix to better explain how to use kube-state-metrics on K8s v1.17 & v1.18.

**Which issue(s) this PR fixes**:
Fixes #1507.

### CODE REVIEW

**Comment by stevehipwell** [comment]
@lilic as requested.

File: README.md [hunk 1]
```diff
@@ -72,10 +72,11 @@ At most, 5 kube-state-metrics and 5 [kubernetes releases](https://github.com/kub
...
 | **v2.1.0**         |         -/✓         |          -/✓         |          ✓           |          ✓           |          ✓           |
 | **master**         |         -/✓         |          -/✓         |          ✓           |          ✓           |          ✓           |
+
 - `✓` Fully supported version range.
 - `-` The Kubernetes cluster has features the client-go library can't use (additional API objects, deprecated APIs, etc).
 
-**Note:** The `v2.0.0-alpha.2+` and `master` releases of kube-state-metrics work on Kubernetes v1.17 and v1.18 excluding Ingress or CertificateSigningRequest resource metrics. If you require those metrics and are on an older Kubernetes version, use v2.0.0-alpha.1 or v1.9.8 kube-state-metrics release.
+**Note:** kube-state-metrics releases from `v2.0.0` work on Kubernetes v1.17 and v1.18 but the Ingress and CertificateSigningRequest resource metrics are missing. If you require those metrics and are on an older Kubernetes version you can use the `v1.9.8` kube-state-metrics release; but please be aware that due to CLI flag changes this image is not compatible with `v2.0.0` and later.
```
Comment (mrueg) [inline README.md:79]:
```suggestion
**Note:** kube-state-metrics releases from `v2.0.0` work on Kubernetes v1.17 and v1.18 but the Ingress and CertificateSigningRequest resource metrics are not available. If you require those metrics and are on an older Kubernetes version you can use the `v1.9.8` kube-state-metrics release; but please be aware that due to CLI flag changes this image is not compatible with `v2.0.0` and later.
```

File: README.md [hunk 2]
```diff
@@ -72,10 +72,11 @@ At most, 5 kube-state-metrics and 5 [kubernetes releases](https://github.com/kub
...
 | **v2.1.0**         |         -/✓         |          -/✓         |          ✓           |          ✓           |          ✓           |
 | **master**         |         -/✓         |          -/✓         |          ✓           |          ✓           |          ✓           |
+
 - `✓` Fully supported version range.
 - `-` The Kubernetes cluster has features the client-go library can't use (additional API objects, deprecated APIs, etc).
 
-**Note:** The `v2.0.0-alpha.2+` and `master` releases of kube-state-metrics work on Kubernetes v1.17 and v1.18 excluding Ingress or CertificateSigningRequest resource metrics. If you require those metrics and are on an older Kubernetes version, use v2.0.0-alpha.1 or v1.9.8 kube-state-metrics release.
+**Note:** kube-state-metrics releases from `v2.0.0` work on Kubernetes v1.17 and v1.18 but the Ingress and CertificateSigningRequest resource metrics are not available. If you require those metrics and are on an older Kubernetes version you can use the `v1.9.8` kube-state-metrics release; but please be aware that due to CLI flag changes this image is not compatible with `v2.0.0` and later.
```
Comment (spaparaju) [inline README.md:79]: The current `kube-state-metrics v2.0.0 +` releases work on Kubernetes v1.17 and v1.18 excluding Ingress or CertificateSigningRequest resource metrics. If you require those metrics on an older Kubernetes version, use `kube-state-metrics v1.9.8` .

File: README.md [hunk 3]
```diff
@@ -324,7 +325,7 @@ For the full list of arguments available, see the documentation in [docs/cli-arg
 
 #### Helm Chart
 
-The official Helm chart is maintained in [prometheus-community/helm-charts](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-state-metrics).
+The official [Helm chart](https://artifacthub.io/packages/helm/prometheus-community/kube-state-metrics/) is maintained in [prometheus-community/helm-charts](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-state-metrics).
```
Comment (spaparaju) [inline README.md:328]: Starting from `kube-state-metrics v2.13.3`, the official Helm chart is maintained at [prometheus-community/helm-charts](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-state-metrics).

**Review by spaparaju — CHANGES_REQUESTED** [review]

**Comment by stevehipwell** [comment]
@spaparaju I've added your changes whilst still keeping one of the important points that I was trying to make around the image compatibility of the Helm chart due to the CLI args changing. This matters where the kube-state-metrics chart is included from another chart (e.g. kube-prometheus-stack) where we can only override the values such as image tag and not the chart version.

**Review by spaparaju — APPROVED** [review]
LGTM 👍

**Review by lilic — COMMENTED** [review]
/lgtm

Thanks!

**Pull request merged**